File: packages/engine/src/cpus/6502/instructions/sbc.test.js
import { test, expect } from 'vitest'
import { BufferMapper } from '../../../buffer_mapper'
import { CPU } from '..'

test('sbc imm', () => {
  const mmu = new BufferMapper()
  const cpu = new CPU(mmu)
  cpu.acc = 2
  cpu.carry = true
  mmu.data = [0xe9, 1] // SBC #$01
  const g = cpu.run()
  for (let i = 0; i < 2; i++) g.next()
  expect(cpu.acc).toBe(1)
  expect(cpu.carry).toBeTruthy()
})

File: packages/engine/src/cores/famibob/cpu/instructions/sbc.test.zig
const testing = @import("../testing/setup.zig");

test "SBC imm" {
    var cpu = testing.createCPU();
    cpu.acc = 2;
    cpu.c = true;
    cpu.write(0, 0xe9);
    cpu.write(1, 1);
    for (0..2) |_| {
        cpu.process();
    }
    try testing.expectEqual(1, cpu.acc);
    try testing.expectEqual(true, cpu.c);
}
